pagopa/pn-frontend
# Pn 2125 FE gestione disservizi api sui ToS destinatario

## Short description
When a user enters either pf-webapp or pa-webapp, an API call is made in order to ascertain whether the user has accepted the ToS or not; the called endpoint is `GET user-consents/v1/consents/TOS`.  
If the call *fails* (i.e. it doesn't respond "the user hasn't accepted the ToS", but responds with an error status), then an automatic logout is performed, showing the user an ad-hoc message in a modal before redirecting to the login.

Previously, in such a case the user was asked to (eventually again) accept the ToS.

## List of changes proposed in this pull request
The `SessionGuard` now checks the existence of errors related to the API call to retrieve the ToS status, and reacts in consequence.  
There are two entries added to the `common.json` file in both pf-webapp and pa-webapp, for the ad-hoc contents of the logout modal.

I also restated the setting of the initial page for pa-webapp depending on whether the user has accepted the ToS or not, this was lost from the changes for [PN-2313](https://pagopa.atlassian.net/browse/PN-2313) when merged a version of `develop` in which `VerifyUser` was replaced by `SessionGuard`. I also made the sidemenu not to appear if the user hasn't accepted the ToS.

## How to test
For either pf-webapp or pa-webapp, you should be able to make the ToS status API call fail, in that case just login to the desired app. The user shouldn't access the application, but instead a modal should be shown and then the user should be redicted to login.
Moreover, for pa-webapp you should be able to access with a user which has not accepted the ToS. You should note that: the sidemenu is not present, and any attempt to access directly to a URL like /dashboard from the browser URL field end in the /tos page. Conversely, any attempt to access /tos made by a user which has accepted the ToS would end in /dashboard.

The next-to-last commit includes commented lines in `src/api/consents/Consents.api.ts` (both pf-webapp and pa-webapp) that makes the call either fail or response that the user hasn't accepted the ToS.
